scripts/pages/leaderboards/leaderboards.ts

```diff
@@ -152,16 +181,75 @@ class LeaderboardsHandler {
 	/**
 	 * Hide the button to go to the end of run page.
 	 */
-	onMapLoad(_isOfficial: boolean) {
+	onMapLoad(isOfficial: boolean) {
 		this.panels.endOfRunButton.visible = false;
+
+		if (isOfficial) {
+			return; // Load official leaderboard tracks instead
+		}
+
+		// Try to load tracks from local zones
+		const mapZoneData = MomentumTimerAPI.GetActiveZoneDefs();
+		if (mapZoneData) {
+			// Main track
+			{
+				const trackStr = $.Localize('#Leaderboards_Tracks_Main');
+				const item = $.CreatePanel('Label', this.panels.tracksDropdown, '', {
+					text: trackStr,
+					value: 0
+				});
+				item.SetAttributeInt('trackNum', 1);
+				item.SetAttributeInt('trackType', TrackType.MAIN);
+
+				this.panels.tracksDropdown.AddOption(item);
+			}
+
+			// Stage tracks
+			const segments = mapZoneData.tracks.main.zones.segments;
+			if (segments.length > 1) {
+				segments.forEach((_, index) => {
+					const trackStr = `${$.Localize('#Leaderboards_Tracks_Stage')} ${index + 1}`;
+					const item = $.CreatePanel('Label', this.panels.tracksDropdown, '', {
+						text: trackStr,
+						value: index + 1
+					});
+					item.SetAttributeInt('trackNum', index + 1);
+					item.SetAttributeInt('trackType', TrackType.STAGE);
+
+					this.panels.tracksDropdown.AddOption(item);
+				});
+			}
+
+			// Bonus tracks
+			const bonuses = mapZoneData.tracks.bonuses;
+			if (bonuses) {
+				bonuses.forEach((_, index) => {
+					const trackStr = `${$.Localize('#Leaderboards_Tracks_Bonus')} ${index + 1}`;
+					const item = $.CreatePanel('Label', this.panels.tracksDropdown, '', {
+						text: trackStr,
+						value: index + 1
+					});
+					item.SetAttributeInt('trackNum', index + 1);
+					item.SetAttributeInt('trackType', TrackType.BONUS);
+
+					this.panels.tracksDropdown.AddOption(item);
+				});
+			}
+		}
+
+		this.initTracksDropdown();
 	}
 
 	onMapLeaderboardsLoad(map: MMap) {
 		this.panels.tracksDropdown.RemoveAllOptions();
 
-		const currentMode = GameModeAPI.GetCurrentGameMode();
+		const isTabMenu = this.panels.cp.id === 'TabMenuLeaderboard';
+		const currentMode = isTabMenu ? GameModeAPI.GetCurrentGameMode() : GameModeAPI.GetMetaGameMode();
 		map.leaderboards
 			.filter((leaderboard) => leaderboard.gamemode === currentMode)
+			.sort((a, b) =>
+				sortTrackId({ type: a.trackType, number: a.trackNum }, { type: b.trackType, number: b.trackNum })
+			)
```

Comment:
Having to do these `{ type: a.trackType, number: a.trackNum }` objects is *very* ugly, why do these need to be `TrackID`s anyway? I don't know when we'd need to sort a trackID like this (something that comes from timer metadata), it's much more likely we need to sort leaderboards, as in this case. Maybe `sortTrackId` (which I suggested) is a bad name, how about 
```ts
export function sortLeaderboard(
  left: { trackNum: number, trackType: TrackType },
  right:  { trackNum: number, trackType: TrackType ): number {
  return left.trackType === right.trackType ? left.trackNum - right.trackNum : left.trackType - right.trackType
} 
``` 
using an object type so any object that contains `{ trackNum: number, trackType: TrackType }` satisfies the type.